src/swap/utils/getTransactionType.ts

```diff
@@ -0,0 +1,21 @@
+import type { Call } from '../../transaction/types';
+
+export function getTransactionType(
+  transactions: Call[],
+  index: number,
+): 'ERC20' | 'Permit2' | 'Swap' | null {
+  // Permit2 has 3 transactions, 2nd to last is the `Permit2` approval
+  if (transactions.length === 3 && index === 0) {
+    return 'Permit2';
+  }
+  // 2nd to last transaction is the `ERC20` approval
+  if (index === transactions.length - 2) {
+    return 'ERC20';
+  }
+  // Last transaction is the swap itself
+  if (index === transactions.length - 1) {
+    return 'Swap';
+  }
```

Comment:
What about `batch`?

Reply:
we handle batch differently - it will return type `Batched` with a `callsId` field.

Reply:
It is a little weird to need to infer the type from the position in the array when we know the type when adding it to the array.  If something changed in processSwap seems like this could get out of sync.  Could possibly create a transactionTypes array at the same time as the transactions array and reference those types by index in sendSingle.  It is nice to be able to directly pass the transactions without modification.

Reply:
i agree with Adam here but would be okay if you add as fast follow